frontend/src/components/Chat.tsx

```diff
@@ -10,75 +16,101 @@ const styles = {
   }
 };
 
+interface Message {
+  user: string;
+  message: string;
+}
+
 interface Props {
+  classes: any;
   sendMessage: Function;
-  signIn: Function;
+  messages: Message[];
 }
 
 interface State {
   message: string;
-  signedIn: boolean;
 }
 
 class Chat extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = {
-      message: "",
-      signedIn: false
+      message: ""
     };
     this.onChange = this.onChange.bind(this);
     this.enterPressed = this.enterPressed.bind(this);
   }
 
-  onChange = (event: React.FormEvent<HTMLInputElement>) => {
+  onChange = (event: ChangeEvent<HTMLInputElement>) => {
     this.setState({ message: event.currentTarget.value });
   };
 
   enterPressed = (event: React.KeyboardEvent) => {
     event.preventDefault();
     const code = event.keyCode || event.which;
     if (code === 13) {
-      if (this.state.signedIn) {
-        this.props.sendMessage(this.state.message);
+      this.props.sendMessage(this.state.message);
+      this.setState({ message: "" });
+    }
+  };
+
+  renderChat = () => {
+    const chat = [];
+    let count = 0;
+    const setUser = (message: Message) => {
+      if (message.user) {
+        return message.user + ": ";
       } else {
-        this.props.signIn(this.state.message);
-        if (this.state.message) {
-          this.setState({ signedIn: true });
-        }
+        return "";
       }
-      this.setState({ message: "" });
+    };
+
+    for (const m of this.props.messages) {
+      chat.push(
+        <span>
+          <ListItemText style={{ paddingLeft: "10px" }} key={count} primary={setUser(m) + m.message} />
+          <Divider variant="middle" />
+        </span>
+      );
+      count++;
+    }
+    if (this.props.messages.length) {
+      return <List>{chat}</List>;
+    } else {
+      return "";
     }
   };
 
   render() {
-    const signInWindow = (
-      <div>
-        <input
-          placeholder="What is your name?"
-          onChange={this.onChange}
-          onKeyUp={this.enterPressed}
-          value={this.state.message}
-        />
-      </div>
-    );
+    const { classes } = this.props;
+
     const chatWindow = (
       <div>
-        <input
-          placeholder="Send a message"
+        <TextField
+          InputProps={{ className: classes.textField }}
+          InputLabelProps={{ className: classes.textField }}
+          label="Send a Message!"
           onChange={this.onChange}
           onKeyUp={this.enterPressed}
           value={this.state.message}
         />
       </div>
     );
     return (
-      <div style={styles.chatBox}>
-        <h3>Chat:</h3>
-        {this.state.signedIn ? chatWindow : signInWindow}
+      <div style={{ paddingLeft: "2vw" }}>
+        {chatWindow}
```

Comment:
I think it feels more natural to put the chat field below where the messages are displayed